src/lib.rs

```diff
@@ -208,12 +294,72 @@ impl fmt::Display for Error {
                 ref command,
                 ref output,
             } => {
-                write!(
+                let crate_name =
+                    env::var("CARGO_PKG_NAME").unwrap_or(String::from("<NO CRATE NAME>"));
+
+                writeln!(f, "")?;
+
+                // Give a short explanation of what the error is
+                writeln!(
                     f,
-                    "`{}` did not exit successfully: {}\nerror: could not find system library '{}' required by the '{}' crate\n",
-                    command, output.status, name, env::var("CARGO_PKG_NAME").unwrap_or_default(),
+                    "pkg-config {}",
+                    match output.status.code() {
+                        Some(code) => format!("exited with status code {}", code),
+                        None => "was terminated by signal".to_string(),
+                    }
                 )?;
-                format_output(output, f)
+
+                // Give the command run so users can reproduce the error
+                writeln!(f, "> {}\n", command)?;
+
+                // Explain how it was caused
+                writeln!(
+                    f,
+                    "The system library `{}` required by crate `{}` was not found.",
+                    name, crate_name
+                )?;
+                writeln!(
+                    f,
+                    "The file `{}.pc` needs to be installed and the PKG_CONFIG_PATH environment variable must contain its parent directory.",
+                    name
+                )?;
+
+                // There will be no status code if terminated by signal
+                if let Some(_code) = output.status.code() {
+                    // NixOS uses a wrapper script for pkg-config that sets the custom
+                    // environment variable PKG_CONFIG_PATH_FOR_TARGET
+                    let search_path =
+                        if let Ok(path_for_target) = env::var("PKG_CONFIG_PATH_FOR_TARGET") {
+                            Some(path_for_target)
+                        } else if let Ok(path) = env::var("PKG_CONFIG_PATH") {
+                            Some(path)
+                        } else {
+                            None
+                        };
+
+                    // Guess the most reasonable course of action
+                    let hint = if let Some(search_path) = search_path {
+                        writeln!(
+                            f,
+                            "PKG_CONFIG_PATH contains the following:\n{}",
```

Comment:
Isn't this confusing for NixOS users? Maybe remember from which variable the search path comes and print that here.

Looks good to me otherwise, thanks a lot :)

Reply:
Ah, good catch! Will write a quick fix.